keras_retinanet/models/retinanet.py

```diff
@@ -273,10 +301,54 @@ def retinanet(
     if submodels is None:
         submodels = default_submodels(num_classes, num_anchors)
 
-    C3, C4, C5 = backbone_layers
+    if pyramid_levels is None:
+        pyramid_levels = [3,4,5,6,7]
+
+
+    C2, C3, C4, C5 = None, None, None, None
+
+    backbone_layer_counter = 0
+    backbone_layer_number = len(backbone_layers)
+
 
+    if backbone_layer_number < 3:
+        raise ValueError('Must provide at least 3 layers for backbone')
+
+    if backbone_layer_number < 4 and 2 in pyramid_levels:
+        raise ValueError("backbone model providing {} layers. Need 4 backbone layers to use pyramid level 2 layers.\
+         Only resnet returns 4 layers and will work with pyramid level 2 ".format(backbone_layer_number))
```

Comment:
I would leave this detail out, otherwise we need to keep updating this message when other models support it too.

Reply:
alright done.